File: planning-idsdp-rms-ti-backend/src/main/java/com/ibcs/idsdp/trainninginstitute/services/GuarantorService.java
package com.ibcs.idsdp.trainninginstitute.services;

import com.ibcs.idsdp.trainninginstitute.model.domain.GuarantorModel;
import com.ibcs.idsdp.trainninginstitute.web.dto.request.GuarantorRequest;
import com.ibcs.idsdp.trainninginstitute.web.dto.request.UploadGuarantorFileReguest;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.ApiMessageResponse;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.PaginationResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface GuarantorService {
    ResponseEntity<ApiMessageResponse> addGuarantor(GuarantorRequest guarantorRequest);

    ResponseEntity<PaginationResponse<List<GuarantorModel>>> getGuarantor(int pageNo, int pageSize, String guarantorName);

    ResponseEntity<ApiMessageResponse> updateGuarantorById(Long guarantorId, GuarantorRequest guarantorRequest);

    ResponseEntity<ApiMessageResponse> deleteGuarantorById(Long guarantorId);

    ResponseEntity<GuarantorModel> getGuarantorById(Long guarantorId);

    ResponseEntity<GuarantorModel> getGuarantorByProposalId(Long proposalId);

    ResponseEntity<ApiMessageResponse> uploadFile(Long guarantorId, UploadGuarantorFileReguest uploadGuarantorFileReguest);
}

File: planning-idsdp-rms-frontend/src/app/main/modules/training-institute/services/guarantor.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {PaginationResponse} from "../models/pagination-response.model";
import {GuarantorResponse} from "../models/guarantor-response.model"
import {GuarantorRequest, UploadFile} from "../models/guarantor-request.model"
import {environment} from "../../../../../environments/environment";
import {FileUploadModel} from "../../../shared/model/file-upload.model";
@Injectable({
  providedIn: 'root'
})
export class GuarantorService {

    baseTiEndPoint = environment.ibcs.tiBackend;

    constructor(private http: HttpClient) {
    }

    uploadFile(em: UploadFile, guarantorId: number) {
        return this.http.put(this.baseTiEndPoint + "guarantor-agreement/file-upload/"+ guarantorId, em);
    }

    getGuarantorList(size: number, page: number, name: string) {
        return this.http.get<PaginationResponse<any[]>>(
            this.baseTiEndPoint + "guarantor-agreement?pageNo=" + page + "&pageSize=" + size + "&guarantorName=" + name);
    }


    getGuarantors(size: number, page: number) {
        return this.http.get<PaginationResponse<GuarantorResponse[]>>(
            this.baseTiEndPoint + "guarantor-agreement?pageNo=" + page + "&pageSize=" + size);
    }

    addGuarantor(guarantorRequest: GuarantorRequest) {
        return this.http.post(this.baseTiEndPoint + "guarantor-agreement", guarantorRequest);
    }

    deleteGuarantor(guarantorId: number) {
        return this.http.delete(this.baseTiEndPoint + "guarantor-agreement/" + guarantorId);
    }

    getGuarantorById(guarantorId: number) {
        return this.http.get<any>(
            this.baseTiEndPoint + "guarantor-agreement/" + guarantorId);
    }

    editGuarantor(guarantorRequest: GuarantorRequest, guarantorId: number) {
        return this.http.put(this.baseTiEndPoint + "guarantor-agreement/" + guarantorId, guarantorRequest);
    }

    getGuarantorByProposalId(id: number) {
        return this.http.get<GuarantorResponse>(
            this.baseTiEndPoint + "guarantor-agreement/proposal/" + id);
    }
}
